packages/node/src/node.ts

```diff
@@ -101,58 +111,56 @@ export class Node {
             "Reached OP_SIGN middleware without generated commitment."
           );
         }
-        context.signature = {
-          v: -1,
-          r: HashZero,
-          s: HashZero
-        };
-        // TODO: Replace the above hack with the below
-        // context.signature = this.signer.signDigest(
-        //   context.commitment.hashToSign()
-        // );
+
+        context.signature = this.signer.signDigest(
+          context.commitment.hashToSign()
+        );
+
         next();
       }
     );
-  }
 
-  private registerIoMiddleware() {
-    this.instructionExecutor.register(
+    instructionExecutor.register(
       Opcode.IO_SEND,
       async (message: ProtocolMessage, next: Function, context: Context) => {
-        await this.messagingService.send(context.outbox[0].toAddress, {
-          from: this.address,
-          event: NODE_EVENTS.PROTOCOL_MESSAGE_EVENT,
-          data: context.outbox[0]
-        });
+        const from = getAddress(this.address);
+        const to = getAddress(context.outbox[0].toAddress);
+        const data = context.outbox[0];
+
+        if (context.outbox[0].seq === 1) {
```

Comment:
Can you put a brief comment here to highlight how seq 1 is relevant to needing a `Deferred` promise?

I can see it being confusing if context isn't provided.

Reply:
This actually reminds me of something I was going to do. As-is, this code only works for single round-trip message exchanges; it will not work for the Install Virtual App Protocol.